dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/AsyncRpcResult.java

```diff
@@ -144,7 +148,12 @@ public Result getAppResponse() {
             // This should never happen;
             logger.error("Got exception when trying to fetch the underlying result from AsyncRpcResult.", e);
         }
-        return new AppResponse();
+
+        ConsumerModel consumerModel = ApplicationModel.getConsumerModel(invocation.getTargetServiceUniqueName());
+        String methodName = invocation.getMethodName();
+        String params = ReflectUtils.getDesc(invocation.getParameterTypes());
+        MethodDescriptor method = consumerModel.getServiceModel().getMethod(methodName, params);
+        return new AppResponse(ReflectUtils.defaultReturn(method.getReturnClass()));
```

Comment:
why do you need a default value? just null is it ok?

Reply:
when the return type is primitive type, it will cause class cast exception in proxy object.